core/test/matrix/CMakeLists.txt

```diff
@@ -5,6 +5,15 @@ ginkgo_create_test(csr_builder)
 ginkgo_create_test(dense)
 ginkgo_create_test(diagonal)
 ginkgo_create_test(ell)
+add_library(test_fbcsr_sample fbcsr_sample.cpp)
+target_include_directories(test_fbcsr_sample
+  PRIVATE
+  "$<BUILD_INTERFACE:${Ginkgo_SOURCE_DIR}/include>"
+  "$<BUILD_INTERFACE:${Ginkgo_SOURCE_DIR}>"
+  "$<BUILD_INTERFACE:${Ginkgo_BINARY_DIR}/include>"
+  )
```

Comment:
It will be nice to separate the implementation in case I need the test matrices again for other tests. Is it preferable to make it header-only?

Reply:
Doing the library registration here sort of breaks the new algorithm script I'm fairly sure. In addition, so far we always do matrix generation etc inside headers so that shouldn't be a problem. If you need to generate complicated data you could always generate it on CPU before moving it to GPU or anything else as well, or use some kernels, whichever way you would prefer. Both of which should work fine without an external library.

Reply:
This file also creates compilation issues since you do not compile this with C++14 (and maybe some other aspects) like the rest of the library requires.